File: 121-BestTimeToBuyAndSellStock/121-BestTimeToBuyAndSellStock.py
# Last updated: 4/14/2026, 12:48:51 PM
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        l,r=0,1
        maxP=0
        while r!=len(prices):
            if prices[l]<prices[r]:
                prof=prices[r]-prices[l]
                maxP=max(maxP,prof)
            else:
                l=r
            r+=1
        return maxP 

File: 121-BestTimeToBuyAndSellStock/121-BestTimeToBuyAndSellStock.js
// Last updated: 4/14/2026, 12:48:48 PM
/**
 * @param {number[]} prices
 * @return {number}
 */
var maxProfit = function(prices) {
    let l = 0
    let r = 0
    let maxP=0

    while(r!=prices.length){
        if(prices[l]<prices[r]){
            let prof=prices[r]-prices[l]
            maxP=Math.max(maxP,prof)
        }else{
            l=r
        } 
        r++
    }return maxP
};